syfertext/token.py

```diff
@@ -7,7 +7,7 @@
 
 
 class Token:
-    def __init__(self, doc, token_meta: "TokenMeta"):
+    def __init__(self, doc, token_meta: "TokenMeta", i: int):
 
```

Comment:
The key is not really needed here. A token object could stand alone without a doc also, and introducing a key is not consistent with that.

Reply:
I thought that `TokenMeta` is the one who may not be linked to a document, but `Token` is:

```Python
class Token:
    def __init__(self, doc, token_meta: "TokenMeta"):
        self.doc = doc
```

I introduced `token.i` so we can retrieve the position of a token in its document, which also enables neighbor retrieval. The functionality is consistent with [`SpaCy`'s](https://spacy.io/api/token#nbor). 

`nbor` needs the relative position of the current token in the document to be able to get its neighbor. I suppose we can get the position by manually searching for the token in the document but that would be slower. 

What do you think?

Reply:
ya you are right, my bad